modules/navier_stokes/src/fvbcs/INSFVTKEDWallFunctionBC.C

```diff
@@ -82,14 +84,16 @@ INSFVTKEDWallFunctionBC::boundaryValue(const FaceInfo & fi) const
   if (y_plus <= 5.0) // sub-laminar layer
   {
     const auto laminar_value = 2.0 * weight * TKE * mu / std::pow(dist, 2);
-    return laminar_value.value();
+    // Additional zero term to make sure new derivatives are not introduced as y_plus changes
+    return laminar_value + 0 * _mu_t(makeElemArg(&_current_elem), state);
```

Comment:
This is adding unnecessary computational burden to the linear solver. Please add a switch between linear and nonlinear if this is needed for the nonlinear solver

Reply:
those are mostly removed